.github/workflows/build_production_image.yml

```diff
@@ -42,38 +47,24 @@ jobs:
         submodules: recursive
         ref: ${{ github.event.inputs.branch }}
     -
-      name: Determine Image Tags
-      id: generate_tag
-      run: |
-        source ${GITHUB_WORKSPACE}/.github/actions/common.sh
-        _tags=""
-        if [[ $GITHUB_EVENT_NAME == "workflow_dispatch" ]]; then
-          _repo=${{ github.event.inputs.repo }}
-          _input_tag=${{ github.event.inputs.tag }}
-          if [[ ${_repo} == "ldmx/pro" ]]; then
-            error "Provided 'ldmx/pro' repo on non-trunk branch."
-            exit 1
-          fi
-          _tags="${_repo}:${_input_tag},${_repo}:sha-${GITHUB_SHA::8}"
-        elif [[ $GITHUB_REF == refs/tags/* ]]; then
-          # if pushing a git tag ==> get the git tag for the docker tag
-          _git_tag=${GITHUB_REF#refs/tags/}
-          _tags="ldmx/pro:$_git_tag"
-          if [[ $_git_tag =~ ^v[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
-            # there was a version released on github
-            #   v<1-3 digits>.<1-3 digits>.<1-3 digits>
-            _tags="$_tags,ldmx/pro:latest"
-          fi
-        elif [[ $GITHUB_REF == refs/heads/* ]]; then
-          #pushing to trunk
-          _tags="ldmx/pro:edge,ldmx/pro:sha-${GITHUB_SHA::8}"
-        else
-          error "Running Determine Image Tags on a commit that isn't a tag or a branch."
-          exit 1
-        fi
-        set_output tags ${_tags}
-        set_output should_push ${_should_push}
-      shell: bash
+      name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        # Default image name for push actions
+        images: ${{ github.event.inputs.repo || 'ldmx/pro' }}
+        # Tag strategies based on the event type
+        tags: |
+          just prep-version
```

Comment:
This is _not_ spawned into a shell, this is just providing configuration to the metadata-action. You need to
```yaml
    - run: just prep-version
```
below this and before the `build-push` action.

Reply:
I'm not sure if this will work out of the box though. `docker/build-push-action` will checkout a fresh copy of the repository if it doesn't have one yet, so we may need to `actions/checkout` above this as well.

Reply:
So now the order is
- actions/checkout
- docker/metadata-action
- just prep-version
- docker/build-push-action

Not sure how to test this besides merging and looking on dockerhub

Reply:
I think merging and watching is a reasonable choice. We can check the `sha-XXXXX` tag and the build logs to see if the `VERSION` was correctly included in the build before making another ldmx-sw release.

Reply:
OK I'll cut a new release now then